torch/csrc/jit/codegen/cuda/executor.cpp

```diff
@@ -1085,19 +1090,23 @@ std::vector<at::Tensor> FusionExecutor::runFusion(
     launch_params_ =
         computeLaunchParams(launch_constraints, expr_eval, warp_size_);
 
-    // Recompile the kernel if the number of threads in the block has increased
-    if (launch_params_.nThreads() > block_size_high_water_mark) {
+    // Recompile the kernel if the number of threads in the block or
+    // maxrregcount has increased
+    if (launch_params_.nThreads() > block_size_high_water_mark ||
+        maxrregcount > maxrregcount_high_water_mark) {
```

Comment:
changed. it makes sense to recompile for potential better performance but not sure about the overhead. if the input shape changes even slightly as it may have a large chance resulting in different value of maxrregcount and lead to a recompile. If we save several copies of the compiled kernel when a new launch para comes we can map it to the best fit instead of recompile. it seems that we only save the most recent one in the current code.